willow/plugins/pillow.py

```diff
@@ -246,4 +254,144 @@ def to_buffer_rgba(self):
         return RGBAImageBuffer(image.size, image.tobytes())
 
 
-willow_image_classes = [PillowImage]
+class PillowAnimatedImage(Image):
+    def __init__(self, frames):
+        self.frames = frames
+
+    @classmethod
+    def check(cls):
+        _PIL_Image()
+
+    @classmethod
+    def is_format_supported(cls, image_format):
+        return is_format_supported(image_format)
+
+    @Image.operation
+    def get_size(self):
+        return self.frames[0].get_size()
+
+    @Image.operation
+    def get_frame_count(self):
+        return len(self.frames)
+
+    @Image.operation
+    def has_alpha(self):
+        return self.frames[0].has_alpha()
```

Comment:
Is it possible that the first frame has no alpha channel, but some other frames do?

Reply:
Interesting question, I wouldn't think it's possible for only part of the animation to have a transparency channel but will check!

Reply:
I've changed this to now check all the frames for alpha, if one or more has alpha then this operation would return `True`.